### ISSUE TITLE
Collision detection for points outside the image not happening due to early AABB check

### ISSUE THREAD

**Issue by thomai-d** [pythonarcade/arcade#832]
Big hitbox areas are very inaccurate.

![hit1](https://user-images.githubusercontent.com/24974298/111953070-68c84a80-8ae6-11eb-895f-458cb96f8913.gif)

![hit2](https://user-images.githubusercontent.com/24974298/111953073-69f97780-8ae6-11eb-8085-57751ea292e3.gif)

Tested with version:
arcade==2.5.5
Shapely==1.7.1

Small repro-code (you can switch hitboxes in player.py):
[repro-code.zip](https://github.com/pythonarcade/arcade/files/6180168/repro-code.zip)

 Related: #753

**Comment by thomai-d** [comment]
**Explanation**
After debugging arcade code, I found out, that to improve performance, a quick collision check is made before the 'real' collision detection algorithm is run.

This quick check relies on the sprites width/height properties, which in my case are zero for the invisible sprite. Therefore, the collision only works within the monkey height's radius.

**Solution**
To make it work, I have to set the invisible sprite's width/height manually to the hitbox's bounding box width/height.

**Suggestion**
Not sure if using a sprite for collision detection is considered 'hacky', but if this is the officially supported method, I would have expected that by using `set_hit_box()` the sprite's collision radius should be calculated based on the hitbox points and not the sprite's width / height.

**Comment by pvcraven** [comment]
Agreed, I think we need to hook into `set_hit_box` so it also will reset height/width. Thanks for the very detailed report.

**Comment by einarf** [comment]
Doing invisible collision with sprite like that is kind of hacky yes, but it's not too unreasonable.

Some random thoughts:
1) It's probably not reasonable that a sprite doesn't have a texture (adds complexity). It can have a dummy texture for sure, and those are easy to make.
2) If the dummy texture needs to be scaled it will also scale the hit box points. That can be a hit weird.
3) Possibly a sprite type could be added for this kind of use.

**Comment by pvcraven** [comment]
This is still a issue with current development, the width/height of the image is used, and if the hitbox extends past that, it is going to be ignored.

**Comment by einarf** [comment]
I looked at his brifely. When it comes to performance it's alll about the cost of calling left, right, top and bottom. These might need to be cached in the sprite to make it viable, but no idea if this reasonable.

Is it even reasonable that points extend past the texture area?

**Comment by pushfoo** [comment]
> Is it even reasonable that points extend past the texture area?

Things like the anomalies from the STALKER series might do something like this to save on texture space: small sprite, large hitbox to set it off.

**Comment by einarf** [comment]
We now have `arcade.get_sprites_in_rect((l, r, b, t), spritelist)` that can used instead of dummy sprite. Still, this doesn't solve the core issue.

**Comment by alejcas** [comment]
> This quick check relies on the sprites width/height properties, which in my case are zero for the invisible sprite. Therefore, the collision only works within the monkey height's radius.

This quick check also makes the collision detection to fail if the hitbox is set to something bigger than the sprite itself.

See https://discord.com/channels/458662222697070613/696054878593744946/1038023494946017320

**Comment by Cleptomania** [comment]
So we have had a big hitbox re-work very recently, and this problem still exists, however we've had a lot of performance improvements, and are able to achieve greater than 20x speedups to the `are_polygons_intersecting` check that this bounding box check tries to ignore via the upcoming `arcade-accelerate` package. Given all of this, I'm tempted to say that we can just remove the check altogether, and always use the full `are_polygons_intersecting` check which will stay true to the hitbox.

**Comment by DragonMoffon** [comment]
Using the new super stripped back Sprite, could we not provide a `BlankSprite` class to solve this issue? Give it a 2x2 invisible texture that can't be changed, just the hitbox and sprite size.

**Comment by pushfoo** [comment]
EDIT: might not be the same issue, refiled here https://github.com/pythonarcade/arcade/issues/2255

TL;DR: It's still happening, and it isn't just big sprites

One of our users [appears to have run into this on Discord today](https://discord.com/channels/458662222697070613/1230878856722776155). Their sprite tiles appear to be 48x48 and the hitbox appears to be clean. See the video below (Converted with `ffmepg -i file.avi file.mp4`):

https://github.com/pythonarcade/arcade/assets/36696816/d0b054f7-e0f2-41db-a8fe-0aaa704ad62b

**Comment by DragonMoffon** [comment]
The original issue has been resolved when hitboxes and sprite scaling was overhauled. The second issue was a flaw in the bounds checking which would cause issues when the test point coincided with a line from the hitbox. This issue appears to have also been solved.

I would like to point out that the original issue was also ill-formed as arcade does not support concave hitboxes by default. How no-one noticed this issue for 3 years I don't know lol. This raises two concerns from me. Firstly, that there is no clear indication in arcade that you can't use concave hitboxes, and second, we really should support convex collisions.

I'm closing this issue, but I will be creating a new one about convex hitboxes.
